src/snowflake/snowpark/mock/connection.py

```diff
@@ -355,7 +357,7 @@ def execute(
 
         res = execute_mock_plan(plan)
         if isinstance(res, TableEmulator):
-            columns = [*res.columns]
+            columns = [unquote_if_quoted(col_name) for col_name in res.columns]
```

Comment:
Could you explain the reasoning behind this change? 🙂

Reply:
I added some context in docstring and comment